[LLVM][CMake] Set LANG to "C" before linker detection (fix issue #62634).

carlo-bramini opened the pull request:
This PR fixes issue  #62634

carlo-bramini committed:
[LLVM][CMake] Set LANG to "C" before linker detection.

This PR fixes issue  #62634

s-barannikov commented:
How about
```
execute_process( 
    COMMAND ${CMAKE_COMMAND} -E env LANG=C ${command}
    ...
)
```
would that solve the issue?

carlo-bramini commented:
Adding `${CMAKE_COMMAND} -E env LANG=C` also worked fine and it fixes the original issue.
Since your solution looks more cleaner, I updated the patch and I added a comment with the explanation of the change.
